src/display/display_utils.js

```diff
@@ -53,12 +53,15 @@ class FilterFactory {
 
   #_defs;
 
+  #baseURL = null;
+
   #document;
 
   #id = 0;
 
   constructor({ ownerDocument = globalThis.document } = {}) {
     this.#document = ownerDocument;
+    this.#baseURL = this.#document?.URL ? new URL(this.#document.URL) : null;
```

Comment:
`baseURI` is the problem here: in the builtin viewer, `baseURI` is `resource://pdf.js/web` which mismatches with the document URI.
So as far as I can tell we must keep the URL . The comparison is firstly made with scheme and path and in the data case the scheme is `data` and the path is `application/data....` so we really need it.

Reply:
So if it becomes a problem (I mean someone complains about that) then we'll see what to do.

Reply:
It's what I did but the absolute url stuff did the job and I overlooked the data case:
https://phabricator.services.mozilla.com/D172223?id=691180

Reply:
Given that this patch won't handle `data:`-URLs well, I *really* think that we should fix this on the Firefox side instead.

As mentioned we already have a bunch of "is built-in PDF Viewer"-checks in the Firefox C++ code, using the existing helper, so I don't understand why we can't add *one more* to address this bug as well; please see https://searchfox.org/mozilla-central/search?q=nsContentUtils%3A%3AIsPDFJS&path=&case=false&regexp=false

Reply:
An other solution is to remove `<base>` from:
https://github.com/mozilla/pdf.js/blob/4a4c5d0fe27cff7295eab05307b38603c1777893/web/viewer-snippet-firefox-extension.html#L2
which is used to compute the url of the filter.
There is a more general discussion on how resolve url in css:
https://github.com/w3c/csswg-drafts/issues/3320

Reply:
> An other solution is to remove `<base>` from:

That seems *extremely* risky, since any number of things in both the viewer and the platform code most likely depends on that being present; can we please *not* consider this here.

Still, my question here is what's so "special" about *this case* that we cannot be allowed to add one more special-case in the platform code?

Reply:
> > An other solution is to remove `<base>` from:
> 
> That seems _extremely_ risky, since any number of things in both the viewer and the platform code most likely depends on that being present; can we please _not_ consider this here.
> 

I agree about the riskyness of removing `<base>`.

> Still, my question here is what's so "special" about _this case_ that we cannot be allowed to add one more special-case in the platform code?

Because ideally it should work without adding such an exception even if we're in a corner case. 

@emilio ^^